### PR TITLE
Folder renames in projects are a separate operation now

### PR DESCRIPTION
**Pull request by gideonthomas**
Related to https://github.com/mozilla/thimble.mozilla.org/pull/1504 and https://github.com/mozilla/brackets/pull/558. This adds a new route that accepts a payload of path mappings i.e. `oldPath: newPath` as an object and updates the relevant files with the new paths. 

This needs to land before the corresponding Thimble and Brackets patches.

### CODE REVIEW

File: api/modules/projects/controller.js [hunk 1]
```diff
@@ -154,6 +155,45 @@ class ProjectsController extends BaseController {
...
+
+          file.set({ path: renamedPaths[oldPath] });
+
+          return file.save(file.changed, {
+            patch: true,
+            method: `update`,
+            transacting: transaction
+          });
+        });
+      }, { concurrency: 3 });
```
Comment (cadecairos) [inline api/modules/projects/controller.js]: why 3?

**Comment by cadecairos** [comment]
LGTM, one question re: concurrency magic number

File: api/modules/projects/controller.js (hunk 1 above)
Comment (gideonthomas) [inline api/modules/projects/controller.js]: coz reasons!!! I honestly chose a random number, feel free to suggest a way to select a number

Comment (cadecairos) [inline api/modules/projects/controller.js]: nah, three's fine.

**Comment by gideonthomas** [comment]
@cadecairos what do you think about me changing the name of the route from `renamefolder` to `updatepaths`? Do you think it makes more semantic sense?

**Comment by cadecairos** [comment]
Now that you mention it, it makes a lot of sense

**Comment by gideonthomas** [comment]
@cadecairos I made the change, am I good to merge?

**Comment by cadecairos** [comment]
R+

**Pull request merged**